tasks/release.py

```diff
@@ -731,7 +734,7 @@ def create_release_branches(ctx, base_directory="~/dd", major_version: int = 7,
     # Strings with proper branch/tag names
     release_branch = current.branch()
 
-    with agent_context(ctx, get_default_branch()):
+    with agent_context(ctx, get_default_branch(major=get_version_major(release_branch)), commit=commit):
```

Comment:
I _think_ that since `commit` is a mandatory argument to this command, it shouldn't be necessary to specify the branch here (according to [this](https://github.com/DataDog/datadog-agent/blob/b6b1c510e74e40e6161b8f11a42c5f3291376c6f/tasks/libs/common/worktree.py#L119)).

```suggestion
    with agent_context(ctx, commit=commit):
```

I also find it confusing that we'd seem to enter a different context here:
https://github.com/DataDog/datadog-agent/blob/7a75d05fc0708c98476ee67f67a39b33b51d6d73/tasks/release.py#L689-L690

Which looks like it could be overriding the place we actually create the branch from?

Reply:
Yeah, that's the hacky part. In fact only the very first `agent_context` is executed, and any subsequent one is skipped so here we stay on the right commit

Reply:
I'm not gonna block the PR on this, but I feel like we may actually get rid of this one `agent_context`: https://github.com/DataDog/datadog-agent/blob/7a75d05fc0708c98476ee67f67a39b33b51d6d73/tasks/release.py#L689

Since `create_and_update_release_branch` is only called from `create_release_branches`, which already sets the right context, we could leave it up to the caller to set up the right context, so that we don't rely on such quirky behavior. We could perhaps even get rid of the `base_branch` argument too to further reduce confusion.

Reply:
This should be resolved now 🙂